darts/tests/models/forecasting/test_baseline_models.py

```diff
@@ -0,0 +1,418 @@
+    LocalForecastingModel,
+)
+from darts.tests.conftest import tfm_kwargs
+from darts.utils import timeseries_generation as tg
+
+logger = get_logger(__name__)
+
+
+icl = 5
+local_models = [
+    (NaiveDrift, {}),
+    (NaiveMean, {}),
+    (NaiveMovingAverage, {}),
+    (NaiveSeasonal, {}),
+]
+global_models = []
+
+
+try:
+    import torch
+
+    from darts.models import GlobalNaiveAggregate, GlobalNaiveDrift, GlobalNaiveSeasonal
+
+    TORCH_AVAILABLE = True
+
+    global_models += [
+        (
+            GlobalNaiveAggregate,
+            {"input_chunk_length": icl, "output_chunk_length": 3, **tfm_kwargs},
+        ),
+        (
+            GlobalNaiveAggregate,
+            {"input_chunk_length": icl, "output_chunk_length": 1, **tfm_kwargs},
+        ),
+        (
+            GlobalNaiveDrift,
+            {"input_chunk_length": icl, "output_chunk_length": 3, **tfm_kwargs},
+        ),
+        (
+            GlobalNaiveDrift,
+            {"input_chunk_length": icl, "output_chunk_length": 1, **tfm_kwargs},
+        ),
+        (
+            GlobalNaiveSeasonal,
+            {"input_chunk_length": icl, "output_chunk_length": 3, **tfm_kwargs},
+        ),
+        (
+            GlobalNaiveSeasonal,
+            {"input_chunk_length": icl, "output_chunk_length": 1, **tfm_kwargs},
+        ),
+    ]
+
+    def custom_mean_valid(x, dim):
+        return torch.mean(x, dim)
+
+    def custom_mean_invalid_out_shape(x, dim):
+        return x[:1]
+
+    def custom_mean_invalid_signature(x):
+        return torch.mean(x, dim=1)
+
+    def custom_mean_invalid_output_type(x, dim):
+        return torch.mean(x, dim=1).detach().numpy()
+
+except ImportError:
+    logger.warning("Torch not installed - will be skipping Torch models tests")
+    TORCH_AVAILABLE = False
+
+    custom_mean_valid = None
+    custom_mean_invalid_out_shape = None
+    custom_mean_invalid_signature = None
+    custom_mean_invalid_output_type = None
+
+
+class TestBaselineModels:
+    np.random.seed(42)
+    if TORCH_AVAILABLE:
+        torch.manual_seed(42)
+
+    @pytest.mark.parametrize(
+        "config", itertools.product(local_models + global_models, [False, True])
+    )
+    def test_fit_predict(self, config):
+        """Tests fit and predict for univariate and multivariate time series."""
+        (model_cls, model_kwargs), is_multivariate = config
+
+        # min train series length for global naive models
+        series = tg.linear_timeseries(length=icl)
+        if is_multivariate:
+            series.stack(series + 100)
+
+        model = model_cls(**model_kwargs)
+
+        # calling fit before predict
+        with pytest.raises(ValueError):
+            model.predict(n=10)
+
+        # calling fit with covariates
+        if isinstance(model, GlobalForecastingModel):
+            err_type = ValueError
+        else:  # for local models, covariates are not part of signature
+            err_type = TypeError
+        with pytest.raises(err_type):
```

Comment:
Should we also check the content of the exception message to make sure it's raised where we think it is?